internal/parsing/parse.go

```diff
@@ -161,6 +182,63 @@ func parseWith(file fs.File, parsers []Parser, groupNumber int, log *zap.Sugared
 	return &finalResults, nil
 }
 
+func checkIfTestIDsAreUnique(parser Parser, tests []v1.Test, cfg Config) (bool, error) {
+	var identityRecipe *v1.TestIdentityRecipe
+	uniqueTestIdentifiers := make(map[string]struct{})
+
+FRAMEWORKS:
+	for framework, frameworkParsers := range cfg.FrameworkParsers {
+		for _, frameworkParser := range frameworkParsers {
+			if reflect.TypeOf(frameworkParser) != reflect.TypeOf(parser) {
+				continue
+			}
+
+			// TODO: What about "other" frameworks (i.e. when we don't match here)
+			if recipe, ok := cfg.IdentityRecipes[framework.String()]; ok {
+				identityRecipe = &recipe
+			}
+
+			break FRAMEWORKS
+		}
+	}
```

Comment:
I think this all is potentially unnecessary - the `finalResults` we grabbed `.Tests` from has `.Framework` on it. Correct me if I'm missing something though 😅

Reply:
Excellent observation, thank you